hat/assets/js/apps/Iaso/domains/instances/components/SpeedDialInstance/SpeedDialInstance.tsx

```diff
@@ -0,0 +1,156 @@
+    } = props;
+    const { data: orgUnitTypeIds } = useGetOrgUnitTypes(formId);
+    const currentUser = useCurrentUser();
+    const hasfeatureFlag = hasFeatureFlag(
+        currentUser,
+        SHOW_LINK_INSTANCE_REFERENCE,
+    );
+    const hasOrgUnitPermission = userHasPermission(
+        'iaso_org_units',
+        currentUser,
+    );
+    const isOrgUnitAlreadyLinked =
+        currentInstance.org_unit.reference_instance_id !== null;
+    console.log('refinstance', currentInstance.org_unit.reference_instance_id);
+    const {
+        org_unit: orgUnit,
+        latitude: formLat,
+        longitude: formLong,
+        altitude: formAltitude,
+    } = currentInstance ?? {};
+
+    const isGpsEqual =
+        hasOrgUnitPermission !== null &&
+        formLat !== null &&
+        formLong !== null &&
+        formAltitude !== null &&
+        formLat === orgUnit?.latitude &&
+        formLong === orgUnit?.longitude &&
+        formAltitude === orgUnit?.altitude;
+
+    const isLinkActionEnabled =
+        hasOrgUnitPermission &&
+        hasfeatureFlag &&
+        formId.toString() === referenceFormId;
+
+    const baseActions = useBaseActions(
+        currentInstance,
+        orgUnitTypeIds as number[], // forcing the type cast as the select in react-query prevent it from being undefined
+        // @ts-ignore
+        reAssignInstance,
+    );
+
+    const editLocationWithInstanceGps =
+        useEditLocationWithGpsAction(currentInstance);
+
+    const deleteRestore = useDeleteAction(currentInstance);
+
+    const lockAction = useLockAction(currentInstance);
+
+    const enketoAction = useEnketoAction(currentInstance);
+
+    const linkOrgUnitAction = useLinkToOrgUnitAction({
+        currentInstance,
+        isOrgUnitAlreadyLinked,
+        formId,
+        referenceFormId: parseInt(referenceFormId, 10),
+    });
+
+    const onActionSelected = action => {
+        if (action.id === 'instanceEditAction' && props.currentInstance) {
+            props.fetchEditUrl(props.currentInstance, window.location);
+        }
+
+        if (action.id === 'instanceDeleteAction' && props.currentInstance) {
+            props.softDelete(props.currentInstance);
+        }
+        if (action.id === 'instanceRestoreAction' && props.currentInstance) {
+            props.restoreInstance(props.currentInstance);
+        }
+    };
+
+    const actions = [...baseActions, deleteRestore];
+
+    if (isGpsEqual) {
+        actions.unshift(editLocationWithInstanceGps);
+    }
+
+    actions.unshift(lockAction, enketoAction);
+
+    if (isLinkActionEnabled) {
+        actions.push(linkOrgUnitAction);
+    }
+
+    return currentInstance?.can_user_modify ? (
+        <SpeedDialInstanceActions
+            actions={actions}
+            onActionSelected={action => onActionSelected(action)}
+        />
+    ) : null;
+};
+
+const MapStateToProps = () => ({});
+const MapDispatchToProps = dispatch => ({
+    ...bindActionCreators(
```

Comment:
You can remove this and dspatch the action directly.
 Btw some of them are not in use anymore i think (fetchInstanceDetail, setCurrentInstance)

Reply:
Like I said in the PR, this is part of refactoring `onActionSelected` which I left out of this PR